### DISCUSSION TITLE
New Syntax!! For Csq3

### DISCUSSION THREAD

**Discussion by AniketKumar2500** [CsqLang/Csq3#18] [Announcements]
![Newsyn](https://user-images.githubusercontent.com/90708238/208306739-91f9f753-a905-46ec-bfa4-b2e6130fd024.png)
How is it?

**Comment by WithSJ** [comment]
That's really impressive

**Comment by WithSJ** [comment]
Hey... I think before futher devlopment of language we need to make some rules

Right now we don't have any direction of this project.

We need rules and principles of our language
So i am thinking 🤔 and working on it..

you have any Idea about this what principles we need for language

**Reply by AniketKumar2500** [reply]
yes that we could make our language close to English so that a non programmer can also understand it.

**Reply by WithSJ** [reply]
Programmer are who can build logics 
I don't think you need to create language for non programmers we should focus on make it easy to remember

When i started my programming journey  that time i faced problems of complex syntax that's why I thinking 🤔 we need more easy syntax.

If you focus on close to English language 
Code look like this 
```
if the firstnumber is equal to the secondnumber then 
    set the sum to 5
end if
```

**We need to focus on reducing unnecessary keywords**

**Comment by AniketKumar2500** [comment]
right so what could be next?

**Reply by WithSJ** [reply]
Give me some time... 
Soon I will explain to you

**Comment by AniketKumar2500** [comment]
Ok any protoype?

**Reply by WithSJ** [reply]
Right now I have an idea soon I will give you written prototype with examples 

To work on it i need time

**Reply by AniketKumar2500** [reply]
OK

**Comment by WithSJ** [comment]
hey @AniketKumar2500  while designing language i am stuck on this 
can we use one keyword in different syntax with their fundamental

```
for var i : range(5)
    statements
end for 
```
and 

```
for condition
    statements
end for

```

use both loop statements with just "for" keyword  i am trying to reduce keywords 
 if it possible so we not need of "while" loop 

**what do you think about this**

**Comment by AniketKumar2500** [comment]
Hello @WithSJ ,
Difference between for and while:
_**The major difference between for loop and the while loop is that for loop is used when the number of iterations is known, whereas execution is done in the while loop until the statement in the program is proved wrong.**_
so let us say that for loop will do all works similar to traditional for loop and while loop so now this will create conflict with those who studied both for and while becuase they will be puzzled that how for is doing the work of while also.

**Reply by WithSJ** [reply]
Right that's why  i think why we create one more keyword
After "for" keyword if is condition it's work as while or if we use var i: range() it's will work as traditional for loop


Actually in my degine i used "loop" keyword

```
loop var i : range(5)
    Statement 
end loop
```
And 

```
loop condition
    Statement 
end loop
```

**Reply by AniketKumar2500** [reply]
Idea is good so i will work on it. 👍🏻

**Reply by AniketKumar2500** [reply]
Keep working on new ideas :)

**Reply by AniketKumar2500** [reply]
but i am wondering why to put a new keyword loop despite we have while and for ?

**Reply by WithSJ** [reply]
for and while both features are available in loop that's i used loop so nobody get confused about for and while

**Reply by AniketKumar2500** [reply]
but if we are looking at keywords for and while both are expressing their uses so still there is no confusion.

**Reply by WithSJ** [reply]
wait... please. I have a purpose for this why do I want the loop keyword 
right now am writing about operators' concepts in 1-2 hours I will post it in the discussion

**Reply by AniketKumar2500** [reply]
ok
